tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+-- Helper module to replicate a set of steps across multiple screen scenarios
+-- Ideas for improvments:
+--  * Export an iterator so the suits can call the screen change step by step
+--    themselves instead of the current all or nothing
+--  * Offer multiple scenarios such as screen add, remove, resize/swap instead
+--    of having to test all corner cases all the time.
+local wibox = require("wibox")
+local surface = require("gears.surface")
+local wallpaper = require("gears.wallpaper")
+local color = require("gears.color")
+local shape = require("gears.shape")
+
+local module = {}
+
+-- Get the original canvas size before it is "lost"
+local canvas_w, canvas_h = screen[1].geometry.width, screen[1].geometry.height
+
+-- Create virtual screens.
+-- The steps will be executed on each set of screens.
+local dispositions = {
```

Comment:
Why is this an array of arrays of *functions` returning screen layouts?
Wouldn't it be easier to get rid of the functions? E.g.
```lua
local dispositions = {
  {
    { x = 0, y = 0, width = math.floor(canvas_w/2), height = canvas_h },
    { x = math.floor(canvas_w/2), y = 0, width = math.ceil(canvas_w/2), height = canvas_h },
  },
  {
etc
  }
}
```